lib/r10k/action/deploy/environment.rb

```diff
@@ -2,29 +2,20 @@
 require 'r10k/deployment'
 require 'r10k/logging'
 require 'r10k/action/visitor'
+require 'r10k/action/base'
 require 'r10k/deployment/write_lock'
 require 'json'
 
 module R10K
   module Action
     module Deploy
-      class Environment
+      class Environment < R10K::Action::Base
 
-        include R10K::Util::Setopts
-        include R10K::Logging
         include R10K::Deployment::WriteLock
 
         def initialize(opts, argv)
-          @opts = opts
-          @argv = argv.map { |arg| arg.gsub(/\W/,'_') }
-          setopts(opts, {
-            :config     => :self,
-            :cachedir   => :self,
-            :puppetfile => :self,
-            :purge      => :self,
-            :trace      => :self
-          })
-
+          super
+          @argv = @argv.map { |arg| arg.gsub(/\W/,'_') }
```

Comment:
Options are handled through `super`; this transformation needs to be done so that normalized environment names are handled consistently when specified on the CLI and generated from `R10K::Deployment`. It was originally a community contributed fix and isn't the most elegant thing, but it works and I don't want to tackle this cleanup at the moment.